### PR TITLE
Inactivity monitoring for UDP server

### LINKED ISSUE #178: Connection stays open when disabling keep-alive
I think the connections might stay open indefinitely in the DTLS listener when using it with `dtls.WithKeepAlive(nil)`. I expect that when disabling keep-alive, we at least would still see the CPU usage and number of goroutines drop back to zero/original after the context duration is expired or something, but after an hour it is still steady high.

We ran into a problem where our server, using `go-coap`, stays on a high CPU usage after clients have finished there connections. At that point there have been e.g. 500 connections, of which the clients are all finished now and don't sent any more messages, and no new incoming connections. I've been trying to debug the issue and found that when I remove `dtls.WithKeepAlive(nil)` from the DTLS server configuration `dtls.NewServer`, so enable keep-alive, the CPU drops back to zero as expected.

I've also seen that the goroutines count stays high and stable when disabling keep-alive and drop back to the original value with keep-alive enabled.

@jkralik Do you have any ideas whether this is indeed the case or maybe something else is going on? I'm looking into this further myself as well, but I'm not as familiar with the code of course.

### PR DESCRIPTION
**Pull request by Niondir**
Fixes #178

It's only for UDP yet, DTLS is missing yet.

I also updated the connection cleanup test to count go-routines to be sure everything is cleaned up.

### CODE REVIEW

File: udp/client/clientconn.go [hunk 1]
```diff
@@ -469,6 +470,7 @@ func (cc *ClientConn) Sequence() uint64 {
 }
 
 func (cc *ClientConn) Process(datagram []byte) error {
+	cc.lastActivity = time.Now()
```
Comment (jkralik) [inline udp/client/clientconn.go:473]: Pls use atomic.Value to store lastActivity and move it to inactivityMonitor struct.

**Comment by jkralik** [comment]
Pls could you make it more generic (creating package net/monitor/inactivity) and use it in tcp/udp/dtls ?

**Comment by Niondir** [comment]
I also added it for DTLS, does it make sense for TCP at all, here we have real connections?

I will move it to `net/inactivity` with the next commit.

**Comment by jkralik** [comment]
I want to change keepalive implementation to use inactivity monitor - so ping will be fired when inactivity occurs.

**Comment by Niondir** [comment]
I also saw some similarities to the keepalive implementation. Whats missing would be the inactivity monitoring for Clients and TCP which was not needed for this issue.
Also the ClientConn will need more calls to notify for activity. Or maybe even the Session could do the job on writeMessage.

I'm done with this PR now. Feel free to have a final look and merge it.

**Review by jkralik — APPROVED** [review]
Thx for contribution. 👍

**Pull request merged**